SingularityExecutor/src/main/java/com/hubspot/singularity/executor/SingularityExecutorThreadChecker.java

```diff
@@ -113,7 +113,13 @@ private int getNumUsedThreads(SingularityExecutorTaskProcessCallable taskProcess
     if (taskProcess.getTask().getTaskInfo().hasContainer() && taskProcess.getTask().getTaskInfo().getContainer().hasDocker()) {
       try {
         String containerName = String.format("%s%s", configuration.getDockerPrefix(), taskProcess.getTask().getTaskId());
-        dockerPid = Optional.of(dockerClient.inspectContainer(containerName).state().pid());
+        int possiblePid = dockerClient.inspectContainer(containerName).state().pid();
+        if (possiblePid == 0) {
+          LOG.warn(String.format("Container %s has pid 0. Running: (%s). Will not try to get threads", containerName, dockerClient.inspectContainer(containerName).state().running()));
```

Comment:
i'd suggest `"Container %s has pid %s (running: %s). Defaulting to 0 threads running."`